In [6]:
import pandas as pd  ## Libreria pandas
import numpy as np   ## Libreria numpy para estadistica
from pathlib import Path ## Libreria pathlib para trabajar en rutas

pd.set_option("display.max_columns", None) ## Mostrar todas las columnas

In [7]:
## Variables para guardar la ubicación de los archivos

comtrade_path = Path("../data/raw/un_comtrade_coffee_cocoa_2020_2025.csv")
population_path = Path("../data/raw/worldbank_population/wb_population_data.csv")
gdp_path = Path("../data/raw/worldbank_gdp/wb_gdp_per_capita_data.csv")

In [ ]:
## Comprobación que la ruta de los archivos es correcta y que existen los archivos

print("Comtrade exists:", comtrade_path.exists())
print("Population exists:", population_path.exists())
print("GDP exists:", gdp_path.exists())

Comtrade exists: True
Population exists: True
GDP exists: True


In [24]:
# Carga de los datasets procedentes de las fuentes oficiales.
# UN Comtrade proporciona los datos de comercio internacional.
# World Bank proporciona los indicadores demográficos y económicos.
#
# Los archivos de World Bank contienen varias líneas iniciales de metadatos
# que no forman parte de la tabla de datos, por lo que se omiten mediante
# el parámetro skiprows=4.
#
# El archivo de UN Comtrade utiliza una codificación distinta de UTF-8,
# por lo que se especifica explícitamente la codificación latin1 para evitar
# errores de lectura.
#
# Además, el CSV de UN Comtrade contiene una coma adicional al final de cada fila.
# Para evitar que Pandas interprete incorrectamente la primera columna como índice
# y desplace el resto de columnas, se utiliza index_col=False.

comtrade = pd.read_csv(
    comtrade_path,
    encoding="latin1",
    index_col=False
)
population = pd.read_csv(population_path, skiprows=4)
gdp = pd.read_csv(gdp_path, skiprows=4)

In [25]:
# Inspección inicial de los datasets.
#
# Este paso forma parte de la validación inicial de los datos
# antes de realizar cualquier limpieza o transformación.
#
# shape devuelve el número de filas y columnas de cada tabla,
# permitiendo comprobar que los archivos se han cargado correctamente
# y conocer su tamaño antes de comenzar el análisis.
#
# head() muestra las primeras filas de cada dataset para revisar:
# - la estructura de las columnas
# - el tipo de información disponible
# - posibles problemas de carga o formato
# - la presencia de valores nulos o columnas innecesarias


print("Comtrade:", comtrade.shape)
print("Population:", population.shape)
print("GDP:", gdp.shape)

display(comtrade.head())
display(population.head())
display(gdp.head())

Comtrade: (2485, 47)
Population: (266, 71)
GDP: (266, 71)


,typeCode,freqCode,refPeriodId,refYear,refMonth,period,reporterCode,reporterISO,reporterDesc,flowCode,flowDesc,partnerCode,partnerISO,partnerDesc,partner2Code,partner2ISO,partner2Desc,classificationCode,classificationSearchCode,isOriginalClassification,cmdCode,cmdDesc,aggrLevel,isLeaf,customsCode,customsDesc,mosCode,motCode,motDesc,qtyUnitCode,qtyUnitAbbr,qty,isQtyEstimated,altQtyUnitCode,altQtyUnitAbbr,altQty,isAltQtyEstimated,netWgt,isNetWgtEstimated,grossWgt,isGrossWgtEstimated,cifvalue,fobvalue,primaryValue,legacyEstimationFlag,isReported,isAggregate
0,C,A,20200101,2020,52,2020,8,ALB,Albania,M,Import,0,W00,World,0,W00,World,H5,HS,True,901,"Coffee, whether or not roasted or decaffeinate...",4,False,C00,TOTAL CPC,0,0,TOTAL MOT,-1,NaN,0.000,False,-1,NaN,0.000,False,0.000,False,0.0,False,3.086732e+07,0.0,3.086732e+07,0,False,True
1,C,A,20200101,2020,52,2020,20,AND,Andorra,M,Import,0,W00,World,0,W00,World,H5,HS,True,901,"Coffee, whether or not roasted or decaffeinate...",4,False,C00,TOTAL CPC,0,0,TOTAL MOT,8,kg,491576.000,False,8,kg,491576.000,False,491576.000,False,0.0,False,5.042072e+06,NaN,5.042072e+06,0,False,True
2,C,A,20200101,2020,52,2020,24,AGO,Angola,M,Import,0,W00,World,0,W00,World,H5,HS,True,901,"Coffee, whether or not roasted or decaffeinate...",4,False,C00,TOTAL CPC,0,0,TOTAL MOT,8,kg,307820.936,True,-1,NaN,0.000,False,307820.936,True,0.0,False,2.918113e+06,NaN,2.918113e+06,6,False,True
3,C,A,20200101,2020,52,2020,28,ATG,Antigua and Barbuda,M,Import,0,W00,World,0,W00,World,H3,HS,True,901,"Coffee, whether or not roasted or decaffeinate...",4,False,C00,TOTAL CPC,0,0,TOTAL MOT,8,kg,63810.351,True,-1,NaN,0.000,False,63810.351,True,0.0,False,2.889606e+05,NaN,2.889606e+05,6,False,True
4,C,A,20200101,2020,52,2020,31,AZE,Azerbaijan,M,Import,0,W00,World,0,W00,World,H5,HS,True,901,"Coffee, whether or not roasted or decaffeinate...",4,False,C00,TOTAL CPC,0,0,TOTAL MOT,8,kg,223677.625,False,8,kg,223677.625,False,223677.625,False,0.0,False,2.010449e+06,NaN,2.010449e+06,0,False,True


,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,1970,1971,1972,1973,1974,1975,1976,1977,1978,1979,1980,1981,1982,1983,1984,1985,1986,1987,1988,1989,1990,1991,1992,1993,1994,1995,1996,1997,1998,1999,2000,2001,2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025,Unnamed: 70
0,Aruba,ABW,"Population, total",SP.POP.TOTL,54922.0,55578.0,56320.0,57002.0,57619.0,58190.0,58694.0,58990.0,59069.0,59052.0,58950.0,58781.0,58047.0,58299.0,58349.0,58295.0,58368.0,58580.0,58776.0,59191.0,59909.0,60563.0,61276.0,62228.0,62901.0,61728.0,59931.0,59159.0,59331.0,60443.0,62753.0,65896.0,69005.0,73685.0,77595.0,79805.0,83021.0,86301.0,88451.0,89659.0,90588.0,91439.0,92074.0,93128.0,95138.0,97635.0,99405.0,100150.0,100917.0,101604.0,101838.0,102591.0,104110.0,105675.0,106807.0,107906.0,108727.0,108735.0,108908.0,109203.0,108587.0,107700.0,107310.0,107359.0,107995.0,NaN,NaN
1,Africa Eastern and Southern,AFE,"Population, total",SP.POP.TOTL,130075728.0,133534923.0,137171659.0,140945536.0,144904094.0,149033472.0,153281203.0,157704381.0,162329396.0,167088245.0,171984985.0,177022314.0,182126556.0,187524135.0,193186642.0,198914573.0,204802976.0,210680842.0,217074286.0,223974122.0,230792729.0,238043099.0,245822010.0,253644643.0,261458202.0,269450407.0,277621771.0,286067346.0,294498625.0,302939121.0,311748681.0,320442961.0,329082707.0,338324002.0,347441809.0,356580375.0,366138524.0,375646235.0,385505757.0,395750933.0,406156661.0,416807868.0,427820358.0,439173286.0,450928044.0,463076637.0,475606210.0,488580707.0,502070763.0,516003448.0,530308387.0,544737983.0,559609961.0,575202699.0,590968990.0,607123269.0,623369401.0,640058741.0,657801085.0,675950189.0,694446100.0,713090928.0,731821393.0,750491370.0,769280888.0,NaN,NaN
2,Afghanistan,AFG,"Population, total",SP.POP.TOTL,9035043.0,9214083.0,9404406.0,9604487.0,9814318.0,10036008.0,10266395.0,10505959.0,10756922.0,11017409.0,11290128.0,11567667.0,11853696.0,12157999.0,12469127.0,12773954.0,13059851.0,13340756.0,13611441.0,13655567.0,13169311.0,11937581.0,10991378.0,10917982.0,11190221.0,11426852.0,11420074.0,11387818.0,11523298.0,11874088.0,12045660.0,12238879.0,13278974.0,14943172.0,16250794.0,17065836.0,17763266.0,18452091.0,19159996.0,19887785.0,20130327.0,20284307.0,21378117.0,22733049.0,23560654.0,24404567.0,25424094.0,25909852.0,26482622.0,27466101.0,28284089.0,29347708.0,30560034.0,31622704.0,32792523.0,33831764.0,34700612.0,35688935.0,36743039.0,37856121.0,39068979.0,40000412.0,40578842.0,41454761.0,42647492.0,NaN,NaN
3,Africa Western and Central,AFW,"Population, total",SP.POP.TOTL,97630925.0,99706674.0,101854756.0,104089175.0,106388440.0,108772632.0,111246953.0,113795019.0,116444636.0,119203521.0,122086536.0,125072948.0,128176494.0,131449942.0,134911581.0,138569918.0,142337272.0,146258576.0,150402616.0,154721711.0,159166518.0,163762473.0,168585118.0,173255157.0,177880746.0,182811038.0,187889141.0,193104347.0,198485027.0,204062274.0,209566031.0,215178709.0,221191375.0,227246778.0,233360104.0,239801875.0,246415446.0,253207584.0,260297834.0,267506298.0,274968446.0,282780717.0,290841795.0,299142845.0,307725100.0,316588476.0,325663158.0,334984176.0,344586109.0,354343844.0,364358270.0,374790143.0,385360349.0,396030207.0,406992047.0,418127845.0,429454743.0,440882906.0,452195915.0,463365429.0,474569351.0,485920997.0,497387180.0,509398589.0,521764076.0,NaN,NaN
4,Angola,AGO,"Population, total",SP.POP.TOTL,5231654.0,5301583.0,5354310.0,5408320.0,5464187.0,5521981.0,5581386.0,5641807.0,5702699.0,5763685.0,5852788.0,5991102.0,6174262.0,6388528.0,6613367.0,6842947.0,7074664.0,7317829.0,7576734.0,7847207.0,8133872.0,8435607.0,8751648.0,9082983.0,9425917.0,9779120.0,10139450.0,10497858.0,10861291.0,11238562.0,11626360.0,12023529.0,12423712.0,12827135.0,13249764.0,13699778.0,14170973.0,14660413.0,15159370.0,15667235.0,16194869.0,16747208.0,17327699.0,17943712.0,18600423.0,19291161.0,20015279.0,20

,Country Name,Country Code,Indicator Name,Indicator Code,1960,1961,1962,1963,1964,1965,1966,1967,1968,1969,1970,1971,1972,1973,1974,1975,1976,1977,1978,1979,1980,1981,1982,1983,1984,1985,1986,1987,1988,1989,1990,1991,1992,1993,1994,1995,1996,1997,1998,1999,2000,2001,2002,2003,2004,2005,2006,2007,2008,2009,2010,2011,2012,2013,2014,2015,2016,2017,2018,2019,2020,2021,2022,2023,2024,2025,Unnamed: 70
0,Aruba,ABW,GDP per capita (current US$),NY.GDP.PCAP.CD,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6767.559229,8244.045660,10056.261393,11507.217151,12187.536361,13233.990517,13892.605143,14700.959808,16055.287787,16548.717387,16620.954556,17750.009564,18828.087059,19216.197235,20681.023027,20740.132583,21307.248251,21949.485996,23700.631990,24171.837133,24845.658484,26736.308944,28171.909401,25134.771230,24093.140151,25712.384302,25119.665545,25813.571441,26129.839062,27458.220154,27441.550214,28440.041688,30082.158423,30645.890602,22759.807175,26749.329609,30975.998912,35718.753119,39498.594129,NaN,NaN
1,Africa Eastern and Southern,AFE,GDP per capita (current US$),NY.GDP.PCAP.CD,186.089204,186.909053,197.367547,225.400079,208.962717,226.836135,240.911794,243.773593,257.143823,281.578894,276.732924,294.814162,311.465053,389.729024,463.468775,479.078798,468.774522,518.358685,571.617317,634.448116,773.301205,777.627960,725.547511,732.345461,650.346841,554.221296,578.317452,664.791813,704.142066,728.243319,822.402999,864.174086,732.910915,709.323534,700.772126,766.522931,746.787335,767.399412,697.079415,670.424213,706.727261,625.827815,630.512328,815.432296,989.015464,1126.297323,1235.266608,1381.552597,1447.545754,1408.753467,1628.905248,1761.989261,1732.038021,1706.022744,1690.166792,1498.875240,1334.470500,1528.104224,1552.073722,1507.085600,1351.591669,1562.416175,1679.327622,1571.449189,1615.396356,NaN,NaN
2,Afghanistan,AFG,GDP per capita (current US$),NY.GDP.PCAP.CD,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,174.930991,138.706822,178.954088,198.871116,221.763654,254.184249,274.218554,376.223152,381.733238,452.053705,560.621505,606.694676,651.417134,637.087099,625.054942,565.569730,522.082216,525.469771,491.337221,496.602504,510.787063,356.496214,357.261153,413.757895,NaN,NaN,NaN
3,Africa Western and Central,AFW,GDP per capita (current US$),NY.GDP.PCAP.CD,121.936832,127.451040,133.823783,139.004980,148.545883,155.561897,162.107340,144.940804,146.186183,161.917968,218.672141,195.859987,230.068904,280.643967,368.294725,413.366170,480.447260,490.839577,523.781318,624.870707,763.011988,1328.793688,1164.188488,873.979407,738.545691,754.905547,582.885076,583.317520,563.117147,512.477735,593.024667,607.840414,567.153365,574.472939,580.008123,866.731606,1069.590240,1091.831529,1141.486651,523.797280,518.969226,532.972787,619.337335,697.481256,838.213392,999.770023,1234.641856,1405.880188,1666.520483,1453.495284,1662.633658,1844.054367,1941.118243,2131.606091,2221.350338,1860.727686,1630.039439,1574.230564,1720.140092,2216.385055,2030.861659,2112.794076,2138.473153,1841.855064,1411.337029,NaN,NaN
4,Angola,AGO,GDP per capita (current US$),NY.GDP.PCAP.CD,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,729.111965,657.982649,634.221467,636.832811,650.491094,772.468833,697.526602,770.101140,807.439628,907.747893,965.866840,881.919488,668.706017,449.727895,334.973590,404.294818,531.115367,521.702915,429.188127,392.725539,563.733796,533.586202,999.065856,1133.663320,1451.471179,2145.886211,2930.444286,3515.056790,4578.155265,3645.148466,4101.637156,5184.152691,5702.453063,5688.579162,5649.687625,3641.728939,2051.814621,2790.718869,2860.093648,2493.678844,1759.356199,2303.908127,3682.113151,2916.136633,2665.874448,NaN,NaN


In [26]:
# Revisión técnica de la estructura del dataset de comercio internacional.
# info() muestra el número de filas, columnas, tipos de datos y valores no nulos.
# Este paso permite identificar qué variables son numéricas, cuáles son categóricas 
# y qué columnas pueden contener valores nulos o no aportar información útil.

comtrade.info()

<class 'pandas.DataFrame'>
RangeIndex: 2485 entries, 0 to 2484
Data columns (total 47 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   typeCode                  2485 non-null   str    
 1   freqCode                  2485 non-null   str    
 2   refPeriodId               2485 non-null   int64  
 3   refYear                   2485 non-null   int64  
 4   refMonth                  2485 non-null   int64  
 5   period                    2485 non-null   int64  
 6   reporterCode              2485 non-null   int64  
 7   reporterISO               2485 non-null   str    
 8   reporterDesc              2485 non-null   str    
 9   flowCode                  2485 non-null   str    
 10  flowDesc                  2485 non-null   str    
 11  partnerCode               2485 non-null   int64  
 12  partnerISO                2485 non-null   str    
 13  partnerDesc               2485 non-null   str    
 14  partner2Code       

In [ ]:
# Revisión técnica del dataset de población.
# El objetivo es comprobar su estructura, el número de registros disponibles y la presencia de valores nulos por año.
# Esta revisión confirma que 2025 no tiene datos disponibles y que existe
# una columna vacía generada por la descarga: Unnamed: 70.

population.info()

<class 'pandas.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 71 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    str    
 1   Country Code    266 non-null    str    
 2   Indicator Name  266 non-null    str    
 3   Indicator Code  266 non-null    str    
 4   1960            264 non-null    float64
 5   1961            264 non-null    float64
 6   1962            264 non-null    float64
 7   1963            264 non-null    float64
 8   1964            264 non-null    float64
 9   1965            264 non-null    float64
 10  1966            264 non-null    float64
 11  1967            264 non-null    float64
 12  1968            264 non-null    float64
 13  1969            264 non-null    float64
 14  1970            264 non-null    float64
 15  1971            264 non-null    float64
 16  1972            264 non-null    float64
 17  1973            264 non-null    float64
 18  1

In [ ]:
# Revisión técnica del dataset de PIB per cápita.
#
# info() permite comprobar la estructura del dataset, los tipos de datos
# y la disponibilidad de información por año.
#
# Esta revisión servirá para identificar posibles valores nulos y determinar
# qué años y países pueden utilizarse posteriormente en el análisis.

gdp.info()

<class 'pandas.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 71 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    str    
 1   Country Code    266 non-null    str    
 2   Indicator Name  266 non-null    str    
 3   Indicator Code  266 non-null    str    
 4   1960            151 non-null    float64
 5   1961            154 non-null    float64
 6   1962            156 non-null    float64
 7   1963            156 non-null    float64
 8   1964            156 non-null    float64
 9   1965            162 non-null    float64
 10  1966            163 non-null    float64
 11  1967            167 non-null    float64
 12  1968            168 non-null    float64
 13  1969            168 non-null    float64
 14  1970            190 non-null    float64
 15  1971            191 non-null    float64
 16  1972            191 non-null    float64
 17  1973            191 non-null    float64
 18  1

In [29]:
# Validación de las variables principales del dataset de comercio.
#
# Se revisan las columnas que serán utilizadas posteriormente en el análisis:
# - País importador
# - Flujo comercial
# - Producto comercializado
# - Valor de las importaciones
#
# Esta comprobación confirma que el archivo se ha cargado correctamente
# y que las variables clave contienen información coherente.

comtrade[
    [
        "reporterDesc",
        "flowDesc",
        "partnerDesc",
        "cmdDesc",
        "primaryValue"
    ]
].head()

,reporterDesc,flowDesc,partnerDesc,cmdDesc,primaryValue
0,Albania,Import,World,"Coffee, whether or not roasted or decaffeinate...",3.086732e+07
1,Andorra,Import,World,"Coffee, whether or not roasted or decaffeinate...",5.042072e+06
2,Angola,Import,World,"Coffee, whether or not roasted or decaffeinate...",2.918113e+06
3,Antigua and Barbuda,Import,World,"Coffee, whether or not roasted or decaffeinate...",2.889606e+05
4,Azerbaijan,Import,World,"Coffee, whether or not roasted or decaffeinate...",2.010449e+06


In [ ]:
# Revisión de los productos incluidos en el dataset de comercio.
#
# El objetivo es comprobar qué códigos HS se han descargado
# y cuántos registros existen para cada producto.
#
# Esta validación permite confirmar que el dataset contiene café (HS 0901)
# cacao en grano (HS 1801) y preparados de cacao (HS 1806) antes de iniciar la limpieza.

comtrade["cmdCode"].value_counts()

cmdCode
901     875
1806    875
1801    735
Name: count, dtype: int64

In [31]:
# Revisión de las descripciones de producto asociadas a los códigos HS.
#
# Esta comprobación ayuda a interpretar correctamente cada código
# y a construir posteriormente una variable de negocio más legible,
# como product_group o sector.

comtrade["cmdDesc"].value_counts()

cmdDesc
Coffee, whether or not roasted or decaffeinated; husks and skins; coffee substitutes containing coffee in any proportion    875
Chocolate and other food preparations containing cocoa                                                                      875
Cocoa beans; whole or broken, raw or roasted                                                                                735
Name: count, dtype: int64

In [ ]:
# Revisión de los años disponibles en el dataset de comercio.
#
# Esta comprobación permite validar si el periodo descargado contiene
# datos entre 2020 y 2025 y si la cobertura por año es equilibrada.
#
# Es importante porque los datos de importaciones podrán analizarse hasta 2025,
# mientras que los indicadores de World Bank solo estarán disponibles hasta 2024.

comtrade["refYear"].value_counts().sort_index()

# 2025 tiene 205 registros frente a los +/- 400 de los años anteriores, lo que indica que la cobertura de datos para ese año es limitada.

refYear
2020    476
2021    477
2022    474
2023    463
2024    390
2025    205
Name: count, dtype: int64

In [36]:
# Revisión del número de países importadores disponibles en el dataset.
#
# Esta comprobación permite conocer la cobertura geográfica inicial
# antes de filtrar el análisis a países de la Unión Europea.

print(comtrade["reporterDesc"].nunique())

176


In [34]:
# Listado de países importadores incluidos en el dataset.
#
# El dataset descargado contiene más países que los de la Unión Europea,
# por lo que posteriormente será necesario filtrar únicamente los países
# relevantes para el análisis.

sorted(comtrade["reporterDesc"].unique())

['Albania',
 'Algeria',
 'Andorra',
 'Angola',
 'Antigua and Barbuda',
 'Argentina',
 'Armenia',
 'Aruba',
 'Australia',
 'Austria',
 'Azerbaijan',
 'Bahamas',
 'Bahrain',
 'Barbados',
 'Belarus',
 'Belgium',
 'Belize',
 'Benin',
 'Bermuda',
 'Bhutan',
 'Bolivia (Plurinational State of)',
 'Bosnia Herzegovina',
 'Botswana',
 'Brazil',
 'Brunei Darussalam',
 'Bulgaria',
 'Burkina Faso',
 'Burundi',
 'Cabo Verde',
 'Cambodia',
 'Cameroon',
 'Canada',
 'Cayman Isds',
 'Central African Rep.',
 'Chile',
 'China',
 'China, Hong Kong SAR',
 'China, Macao SAR',
 'Colombia',
 'Comoros',
 'Congo',
 'Costa Rica',
 'Croatia',
 'Cuba',
 'Curaçao',
 'Cyprus',
 'Czechia',
 "Côte d'Ivoire",
 'Dem. Rep. of the Congo',
 'Denmark',
 'Djibouti',
 'Dominica',
 'Dominican Rep.',
 'Ecuador',
 'Egypt',
 'El Salvador',
 'Estonia',
 'Eswatini',
 'Ethiopia',
 'Faroe Isds',
 'Fiji',
 'Finland',
 'France',
 'French Polynesia',
 'Gabon',
 'Gambia',
 'Georgia',
 'Germany',
 'Ghana',
 'Greece',
 'Grenada',
 'Guatemal

In [37]:
# Definición de países de la Unión Europea.
#
# El dataset original contiene países de todo el mundo.
# Como el objetivo del proyecto es analizar oportunidades en mercados europeos
# bajo un marco regulatorio comparable, se filtra el análisis a países miembros
# de la Unión Europea.

eu_countries = [
    "Austria",
    "Belgium",
    "Bulgaria",
    "Croatia",
    "Cyprus",
    "Czechia",
    "Denmark",
    "Estonia",
    "Finland",
    "France",
    "Germany",
    "Greece",
    "Hungary",
    "Ireland",
    "Italy",
    "Latvia",
    "Lithuania",
    "Luxembourg",
    "Malta",
    "Netherlands",
    "Poland",
    "Portugal",
    "Romania",
    "Slovakia",
    "Slovenia",
    "Spain",
    "Sweden"
]

In [38]:
# Filtrado del dataset de comercio para conservar únicamente países de la Unión Europea.
#
# Se crea una copia del dataframe filtrado para evitar modificar el dataset original
# descargado desde UN Comtrade.

comtrade_eu = comtrade[
    comtrade["reporterDesc"].isin(eu_countries)
].copy()

print("Dataset original:", comtrade.shape)
print("Dataset UE:", comtrade_eu.shape)
print("Países UE incluidos:", comtrade_eu["reporterDesc"].nunique())

Dataset original: (2485, 47)
Dataset UE: (471, 47)
Países UE incluidos: 27


In [39]:
# Revisión de la cobertura temporal después de filtrar países de la Unión Europea.
#
# Esta comprobación permite ver cuántos registros existen por año
# dentro del mercado objetivo del análisis.

comtrade_eu["refYear"].value_counts().sort_index()

refYear
2020    81
2021    81
2022    81
2023    81
2024    81
2025    66
Name: count, dtype: int64

# El filtrado a países de la Unión Europea deja 471 registros.
# Para el periodo 2020-2024 la cobertura es completa:
# 27 países x 3 productos = 81 registros por año.
#
# En 2025 solo existen 66 registros, por lo que ese año parece incompleto
# en el momento de la descarga. Por este motivo, las métricas comparativas
# principales se calcularán sobre 2020-2024, mientras que 2025 podrá utilizarse
# únicamente como dato reciente con advertencia metodológica.

In [40]:
# Selección de columnas útiles para el análisis.
#
# El dataset original de UN Comtrade contiene muchas columnas técnicas
# que no son necesarias para responder a la pregunta de negocio.
# En esta fase se conservan únicamente las variables necesarias para:
# - identificar país y año
# - identificar producto
# - analizar valor importado
# - analizar volumen importado

trade_clean = comtrade_eu[
    [
        "refYear",
        "reporterISO",
        "reporterDesc",
        "cmdCode",
        "cmdDesc",
        "netWgt",
        "primaryValue"
    ]
].copy()

trade_clean.head()

,refYear,reporterISO,reporterDesc,cmdCode,cmdDesc,netWgt,primaryValue
7,2020,AUT,Austria,901,"Coffee, whether or not roasted or decaffeinate...",6.227891e+07,4.249737e+08
12,2020,BEL,Belgium,901,"Coffee, whether or not roasted or decaffeinate...",1.737792e+08,6.080490e+08
21,2020,BGR,Bulgaria,901,"Coffee, whether or not roasted or decaffeinate...",3.998450e+07,1.313092e+08
33,2020,AUT,Austria,1801,"Cocoa beans; whole or broken, raw or roasted",2.658997e+07,7.387447e+07
38,2020,BEL,Belgium,1801,"Cocoa beans; whole or broken, raw or roasted",8.124282e+07,2.325734e+08


In [41]:
# Renombrado de columnas.
#
# Se sustituyen los nombres técnicos originales por nombres más claros
# y orientados al análisis de negocio.

trade_clean = trade_clean.rename(
    columns={
        "refYear": "year",
        "reporterISO": "country_code",
        "reporterDesc": "country",
        "cmdCode": "commodity_code",
        "cmdDesc": "commodity_description",
        "netWgt": "net_weight_kg",
        "primaryValue": "import_value_usd"
    }
)

trade_clean.head()

,year,country_code,country,commodity_code,commodity_description,net_weight_kg,import_value_usd
7,2020,AUT,Austria,901,"Coffee, whether or not roasted or decaffeinate...",6.227891e+07,4.249737e+08
12,2020,BEL,Belgium,901,"Coffee, whether or not roasted or decaffeinate...",1.737792e+08,6.080490e+08
21,2020,BGR,Bulgaria,901,"Coffee, whether or not roasted or decaffeinate...",3.998450e+07,1.313092e+08
33,2020,AUT,Austria,1801,"Cocoa beans; whole or broken, raw or roasted",2.658997e+07,7.387447e+07
38,2020,BEL,Belgium,1801,"Cocoa beans; whole or broken, raw or roasted",8.124282e+07,2.325734e+08


In [42]:
# Creación de variables de negocio.
#
# Los códigos HS son útiles técnicamente, pero no son intuitivos para una audiencia no técnica.
# Por eso se crean columnas más legibles:
# - sector: diferencia entre café y cacao
# - product_group: clasifica cada producto en una categoría de negocio
#
# Estas variables facilitarán la construcción de filtros y gráficos en el dashboard.

product_mapping = {
    901: {
        "sector": "Coffee",
        "product_group": "Coffee total"
    },
    1801: {
        "sector": "Cocoa",
        "product_group": "Cocoa beans"
    },
    1806: {
        "sector": "Cocoa",
        "product_group": "Chocolate and cocoa preparations"
    }
}

trade_clean["sector"] = trade_clean["commodity_code"].map(
    lambda x: product_mapping[x]["sector"]
)

trade_clean["product_group"] = trade_clean["commodity_code"].map(
    lambda x: product_mapping[x]["product_group"]
)

trade_clean.head()

,year,country_code,country,commodity_code,commodity_description,net_weight_kg,import_value_usd,sector,product_group
7,2020,AUT,Austria,901,"Coffee, whether or not roasted or decaffeinate...",6.227891e+07,4.249737e+08,Coffee,Coffee total
12,2020,BEL,Belgium,901,"Coffee, whether or not roasted or decaffeinate...",1.737792e+08,6.080490e+08,Coffee,Coffee total
21,2020,BGR,Bulgaria,901,"Coffee, whether or not roasted or decaffeinate...",3.998450e+07,1.313092e+08,Coffee,Coffee total
33,2020,AUT,Austria,1801,"Cocoa beans; whole or broken, raw or roasted",2.658997e+07,7.387447e+07,Cocoa,Cocoa beans
38,2020,BEL,Belgium,1801,"Cocoa beans; whole or broken, raw or roasted",8.124282e+07,2.325734e+08,Cocoa,Cocoa beans


In [43]:
# Validación de las nuevas variables de negocio.
#
# Se comprueba que cada código de producto se haya clasificado correctamente
# en su sector y grupo de producto correspondiente.

trade_clean[
    ["commodity_code", "sector", "product_group"]
].drop_duplicates().sort_values("commodity_code")

,commodity_code,sector,product_group
7,901,Coffee,Coffee total
33,1801,Cocoa,Cocoa beans
56,1806,Cocoa,Chocolate and cocoa preparations


In [44]:
# Cálculo del precio medio por kilogramo.
#
# Esta métrica permite comparar el valor económico relativo de cada producto
# más allá del volumen importado.
#
# Se calcula dividiendo el valor total importado entre el peso neto importado.
# Si el peso neto es cero, el resultado se deja como valor nulo para evitar
# divisiones incorrectas.

trade_clean["avg_price_usd_kg"] = np.where(
    trade_clean["net_weight_kg"] > 0,
    trade_clean["import_value_usd"] / trade_clean["net_weight_kg"],
    np.nan
)

trade_clean.head()

,year,country_code,country,commodity_code,commodity_description,net_weight_kg,import_value_usd,sector,product_group,avg_price_usd_kg
7,2020,AUT,Austria,901,"Coffee, whether or not roasted or decaffeinate...",6.227891e+07,4.249737e+08,Coffee,Coffee total,6.823717
12,2020,BEL,Belgium,901,"Coffee, whether or not roasted or decaffeinate...",1.737792e+08,6.080490e+08,Coffee,Coffee total,3.498974
21,2020,BGR,Bulgaria,901,"Coffee, whether or not roasted or decaffeinate...",3.998450e+07,1.313092e+08,Coffee,Coffee total,3.284003
33,2020,AUT,Austria,1801,"Cocoa beans; whole or broken, raw or roasted",2.658997e+07,7.387447e+07,Cocoa,Cocoa beans,2.778283
38,2020,BEL,Belgium,1801,"Cocoa beans; whole or broken, raw or roasted",8.124282e+07,2.325734e+08,Cocoa,Cocoa beans,2.862694


In [ ]:
# Validación estadística inicial de las principales métricas numéricas.
#
# describe() permite revisar mínimos, máximos, medias y dispersión.
# Este paso ayuda a detectar valores extremos, nulos o precios medios poco realistas que puedan requerir limpieza.

trade_clean[
    ["net_weight_kg", "import_value_usd", "avg_price_usd_kg"]
].describe()

,net_weight_kg,import_value_usd,avg_price_usd_kg
count,4.590000e+02,4.710000e+02,452.000000
mean,1.043302e+08,5.490346e+08,6.423919
std,1.911415e+08,9.522093e+08,3.133381
min,0.000000e+00,1.044360e+03,0.724702
25%,4.289968e+06,4.001327e+07,4.568548
50%,3.347903e+07,2.056994e+08,5.992270
75%,9.924274e+07,5.430567e+08,7.729092
max,1.216999e+09,7.809133e+09,27.668914


In [ ]:
# Identificación de registros nulos o igual a cero.
#
# Estos registros impiden calcular el precio medio por kilogramo y pueden
# afectar a determinados análisis comparativos.
#
# La revisión permitirá decidir posteriormente si deben eliminarse,
# imputarse o mantenerse únicamente para análisis de valor comercial.

trade_clean[
    trade_clean["net_weight_kg"].isna() |
    (trade_clean["net_weight_kg"] == 0)
][
    [
        "year",
        "country",
        "commodity_code",
        "product_group",
        "net_weight_kg",
        "import_value_usd"
    ]
]


,year,country,commodity_code,product_group,net_weight_kg,import_value_usd
186,2020,Hungary,901,Coffee total,0.0,1.121816e+08
246,2020,Hungary,1806,Chocolate and cocoa preparations,0.0,3.038766e+08
372,2020,Portugal,1806,Chocolate and cocoa preparations,NaN,2.453777e+08
647,2021,Greece,901,Coffee total,0.0,2.881543e+08
669,2021,Hungary,901,Coffee total,0.0,1.362356e+08
729,2021,Hungary,1806,Chocolate and cocoa preparations,0.0,3.369547e+08
1137,2022,Hungary,901,Coffee total,0.0,1.583410e+08
1199,2022,Hungary,1806,Chocolate and cocoa preparations,0.0,3.424929e+08
1342,2022,Slovakia,1801,Cocoa beans,NaN,2.535563e+07
1485,2023,Belgium,1806,Chocolate and cocoa preparations,NaN,1.110888e+09


# Los registros sin peso neto o con peso igual a cero no permiten calcular
# correctamente el precio medio por kilogramo.
#
# Sin embargo, estos registros sí contienen valor de importación, por lo que
# se mantienen en el dataset para análisis de valor comercial.
#
# La variable avg_price_usd_kg permanecerá como valor nulo en estos casos
# y estos registros se excluirán únicamente de análisis basados en precio/kg.

In [48]:
# Revisión de posibles duplicados.
#
# La unidad de análisis esperada es país, año y producto.
# Por tanto, no debería haber más de un registro para la misma combinación
# de country_code, year y commodity_code.

duplicates = trade_clean.duplicated(
    subset=["country_code", "year", "commodity_code"],
    keep=False
)

print("Número de registros duplicados:", duplicates.sum())

trade_clean[duplicates].sort_values(
    ["country", "year", "commodity_code"]
)

Número de registros duplicados: 0


,year,country_code,country,commodity_code,commodity_description,net_weight_kg,import_value_usd,sector,product_group,avg_price_usd_kg


In [49]:
# Limpieza inicial del dataset de población.
#
# Se elimina la columna vacía generada durante la descarga
# y el año 2025, ya que no contiene datos.
#
# Esta limpieza simplifica las transformaciones posteriores.

population = population.drop(
    columns=["Unnamed: 70", "2025"]
)

In [50]:
# Limpieza inicial del dataset de PIB per cápita.
#
# Se elimina la columna vacía generada durante la descarga
# y el año 2025, ya que no contiene datos disponibles.

gdp = gdp.drop(
    columns=["Unnamed: 70", "2025"]
)

In [ ]:
# Transformación del dataset de población trasponiendo las columnas de año.
#
# El dataset original está en formato ancho (un año por columna).
# Para poder relacionarlo con el dataset de comercio,
# se transforma a formato largo (el año es una columna con valores) utilizando melt().
#
# El resultado tendrá una fila por país y año.

population_long = population.melt(
    id_vars=[
        "Country Name",
        "Country Code",
        "Indicator Name",
        "Indicator Code"
    ],
    var_name="year",
    value_name="population"
)

population_long.head()

,Country Name,Country Code,Indicator Name,Indicator Code,year,population
0,Aruba,ABW,"Population, total",SP.POP.TOTL,1960,54922.0
1,Africa Eastern and Southern,AFE,"Population, total",SP.POP.TOTL,1960,130075728.0
2,Afghanistan,AFG,"Population, total",SP.POP.TOTL,1960,9035043.0
3,Africa Western and Central,AFW,"Population, total",SP.POP.TOTL,1960,97630925.0
4,Angola,AGO,"Population, total",SP.POP.TOTL,1960,5231654.0


In [52]:
# Transformación del dataset de PIB per cápita a formato largo.
#
# Igual que en población, cada año pasa de ser una columna
# a convertirse en un registro independiente.

gdp_long = gdp.melt(
    id_vars=[
        "Country Name",
        "Country Code",
        "Indicator Name",
        "Indicator Code"
    ],
    var_name="year",
    value_name="gdp_per_capita"
)

gdp_long.head()

,Country Name,Country Code,Indicator Name,Indicator Code,year,gdp_per_capita
0,Aruba,ABW,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,NaN
1,Africa Eastern and Southern,AFE,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,186.089204
2,Afghanistan,AFG,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,NaN
3,Africa Western and Central,AFW,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,121.936832
4,Angola,AGO,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,NaN


In [ ]:
# Validación de la transformación a formato largo.
print("Population long:", population_long.shape)
print("GDP long:", gdp_long.shape)

display(population_long.head())
display(gdp_long.head())

Population long: (17290, 6)
GDP long: (17290, 6)


,Country Name,Country Code,Indicator Name,Indicator Code,year,population
0,Aruba,ABW,"Population, total",SP.POP.TOTL,1960,54922.0
1,Africa Eastern and Southern,AFE,"Population, total",SP.POP.TOTL,1960,130075728.0
2,Afghanistan,AFG,"Population, total",SP.POP.TOTL,1960,9035043.0
3,Africa Western and Central,AFW,"Population, total",SP.POP.TOTL,1960,97630925.0
4,Angola,AGO,"Population, total",SP.POP.TOTL,1960,5231654.0


,Country Name,Country Code,Indicator Name,Indicator Code,year,gdp_per_capita
0,Aruba,ABW,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,NaN
1,Africa Eastern and Southern,AFE,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,186.089204
2,Afghanistan,AFG,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,NaN
3,Africa Western and Central,AFW,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,121.936832
4,Angola,AGO,GDP per capita (current US$),NY.GDP.PCAP.CD,1960,NaN


In [58]:
# Filtrado de los datasets auxiliares.
#
# Se conservan únicamente los países de la Unión Europea y los años necesarios para cruzar con el dataset de comercio.
# En World Bank, el año queda como texto tras melt(), por lo que se convierte a entero.

population_long["year"] = population_long["year"].astype(int)
gdp_long["year"] = gdp_long["year"].astype(int)

population_eu = population_long[
    (population_long["Country Name"].isin(eu_countries)) &
    (population_long["year"].between(2020, 2024))
].copy()

gdp_eu = gdp_long[
    (gdp_long["Country Name"].isin(eu_countries)) &
    (gdp_long["year"].between(2020, 2024))
].copy()

print("Population EU:", population_eu.shape)
print("GDP EU:", gdp_eu.shape)

Population EU: (135, 6)
GDP EU: (135, 6)


In [57]:
# Comprobación de países de la Unión Europea no encontrados en World Bank.
#
# El objetivo es detectar diferencias de nomenclatura entre fuentes,
# por ejemplo "Czechia" frente a otro nombre equivalente.

wb_countries = set(population_long["Country Name"].unique())
missing_countries = sorted(set(eu_countries) - wb_countries)

missing_countries

[]

In [56]:
# Normalización de nombres de países entre fuentes.
#
# UN Comtrade y World Bank no siempre utilizan exactamente la misma
# nomenclatura para los países. En este caso, World Bank usa
# "Slovak Republic" mientras que UN Comtrade usa "Slovakia".
#
# Se corrige el nombre para poder cruzar correctamente ambos datasets.

country_name_mapping = {
    "Slovak Republic": "Slovakia"
}

population_long["Country Name"] = population_long["Country Name"].replace(country_name_mapping)
gdp_long["Country Name"] = gdp_long["Country Name"].replace(country_name_mapping)

In [ ]:
# Selección y renombrado de columnas de población.
#
# Se conservan únicamente las variables necesarias para el modelo analítico.
# El resultado será una tabla dimensional relacionada por país y año.

population_final = population_eu[
    [
        "Country Code",
        "Country Name",
        "year",
        "population"
    ]
].copy()

population_final = population_final.rename(
    columns={
        "Country Code": "country_code",
        "Country Name": "country"
    }
)

population_final.head()

,country_code,country,year,population
15974,AUT,Austria,2020,8916864.0
15977,BEL,Belgium,2020,11538604.0
15981,BGR,Bulgaria,2020,6550696.0
16013,CYP,Cyprus,2020,1302247.0
16014,CZE,Czechia,2020,10697858.0


In [60]:
# Selección y renombrado de columnas de PIB per cápita.
#
# Se eliminan las columnas descriptivas que ya no son necesarias
# para el análisis y se adopta la misma nomenclatura utilizada
# en el resto del proyecto.

gdp_final = gdp_eu[
    [
        "Country Code",
        "Country Name",
        "year",
        "gdp_per_capita"
    ]
].copy()

gdp_final = gdp_final.rename(
    columns={
        "Country Code": "country_code",
        "Country Name": "country"
    }
)

gdp_final.head()

,country_code,country,year,gdp_per_capita
15974,AUT,Austria,2020,48716.409890
15977,BEL,Belgium,2020,45906.287581
15981,BGR,Bulgaria,2020,10760.211975
16013,CYP,Cyprus,2020,28130.341797
16014,CZE,Czechia,2020,23472.891545


In [ ]:
# Validación final de los datasets limpios y transformados.
#
# Se comprueba que los datasets de comercio, población y PIB per cápita
# tienen el número esperado de registros y columnas antes de iniciar
# el análisis exploratorio y la construcción del modelo analítico.

print(trade_clean.shape)
print(population_final.shape)
print(gdp_final.shape)

(471, 10)
(135, 4)
(135, 4)


In [62]:
# Integración de comercio con población.
#
# Se cruza el dataset de importaciones con la población utilizando
# país y año como claves comunes.
#
# Se usa left join para conservar todos los registros de comercio
# y añadir población cuando esté disponible.

trade_enriched = trade_clean.merge(
    population_final,
    on=["country_code", "country", "year"],
    how="left"
)

trade_enriched.head()

,year,country_code,country,commodity_code,commodity_description,net_weight_kg,import_value_usd,sector,product_group,avg_price_usd_kg,population
0,2020,AUT,Austria,901,"Coffee, whether or not roasted or decaffeinate...",6.227891e+07,4.249737e+08,Coffee,Coffee total,6.823717,8916864.0
1,2020,BEL,Belgium,901,"Coffee, whether or not roasted or decaffeinate...",1.737792e+08,6.080490e+08,Coffee,Coffee total,3.498974,11538604.0
2,2020,BGR,Bulgaria,901,"Coffee, whether or not roasted or decaffeinate...",3.998450e+07,1.313092e+08,Coffee,Coffee total,3.284003,6550696.0
3,2020,AUT,Austria,1801,"Cocoa beans; whole or broken, raw or roasted",2.658997e+07,7.387447e+07,Cocoa,Cocoa beans,2.778283,8916864.0
4,2020,BEL,Belgium,1801,"Cocoa beans; whole or broken, raw or roasted",8.124282e+07,2.325734e+08,Cocoa,Cocoa beans,2.862694,11538604.0


In [66]:
# Integración del PIB per cápita.
#
# Se añade el indicador económico al dataset enriquecido.
# Al igual que en el cruce anterior, se conserva la tabla de comercio
# como base principal del análisis.

trade_enriched = trade_enriched.merge(
    gdp_final,
    on=["country_code", "country", "year"],
    how="left"
)

trade_enriched.head()

,year,country_code,country,commodity_code,commodity_description,net_weight_kg,import_value_usd,sector,product_group,avg_price_usd_kg,population,gdp_per_capita_x,gdp_per_capita_y,gdp_per_capita
0,2020,AUT,Austria,901,"Coffee, whether or not roasted or decaffeinate...",6.227891e+07,4.249737e+08,Coffee,Coffee total,6.823717,8916864.0,48716.409890,48716.409890,48716.409890
1,2020,BEL,Belgium,901,"Coffee, whether or not roasted or decaffeinate...",1.737792e+08,6.080490e+08,Coffee,Coffee total,3.498974,11538604.0,45906.287581,45906.287581,45906.287581
2,2020,BGR,Bulgaria,901,"Coffee, whether or not roasted or decaffeinate...",3.998450e+07,1.313092e+08,Coffee,Coffee total,3.284003,6550696.0,10760.211975,10760.211975,10760.211975
3,2020,AUT,Austria,1801,"Cocoa beans; whole or broken, raw or roasted",2.658997e+07,7.387447e+07,Cocoa,Cocoa beans,2.778283,8916864.0,48716.409890,48716.409890,48716.409890
4,2020,BEL,Belgium,1801,"Cocoa beans; whole or broken, raw or roasted",8.124282e+07,2.325734e+08,Cocoa,Cocoa beans,2.862694,11538604.0,45906.287581,45906.287581,45906.287581


In [67]:
# Validación del dataset integrado.
#
# Se revisa el tamaño final y la presencia de valores nulos
# en las variables añadidas desde World Bank.

print(trade_enriched.shape)

trade_enriched[
    ["year", "country", "product_group", "population", "gdp_per_capita"]
].isna().sum()

(471, 14)


year               0
country            0
product_group      0
population        66
gdp_per_capita    66
dtype: int64

In [68]:
# Comprobación de los valores nulos tras integrar World Bank.
#
# La ausencia de población y PIB per cápita debería corresponder al año 2025,
# ya que los datasets de World Bank descargados no contienen valores para ese año.

trade_enriched[
    trade_enriched["population"].isna()
]["year"].value_counts()

year
2025    66
Name: count, dtype: int64

In [69]:
# Creación del dataset enriquecido para métricas per cápita.
#
# Se excluye 2025 porque no dispone de población ni PIB per cápita.
# El año 2025 se mantendrá disponible para análisis descriptivo de importaciones,
# pero no para métricas cruzadas con indicadores socioeconómicos.

trade_enriched_complete = trade_enriched[
    trade_enriched["year"].between(2020, 2024)
].copy()

trade_enriched_complete[
    ["population", "gdp_per_capita"]
].isna().sum()

population        0
gdp_per_capita    0
dtype: int64

In [70]:
# Cálculo de métricas per cápita.
#
# Estas variables permiten comparar países de distinto tamaño poblacional.
# Un país grande puede importar más en valor absoluto, pero no necesariamente
# tener mayor intensidad de mercado por habitante.
#
# Se calculan únicamente sobre el dataset 2020-2024, ya que 2025 no dispone
# de población ni PIB per cápita en World Bank.

trade_enriched_complete["import_value_usd_per_capita"] = (
    trade_enriched_complete["import_value_usd"] / trade_enriched_complete["population"]
)

trade_enriched_complete["net_weight_kg_per_capita"] = (
    trade_enriched_complete["net_weight_kg"] / trade_enriched_complete["population"]
)

trade_enriched_complete.head()

,year,country_code,country,commodity_code,commodity_description,net_weight_kg,import_value_usd,sector,product_group,avg_price_usd_kg,population,gdp_per_capita_x,gdp_per_capita_y,gdp_per_capita,import_value_usd_per_capita,net_weight_kg_per_capita
0,2020,AUT,Austria,901,"Coffee, whether or not roasted or decaffeinate...",6.227891e+07,4.249737e+08,Coffee,Coffee total,6.823717,8916864.0,48716.409890,48716.409890,48716.409890,47.659542,6.984397
1,2020,BEL,Belgium,901,"Coffee, whether or not roasted or decaffeinate...",1.737792e+08,6.080490e+08,Coffee,Coffee total,3.498974,11538604.0,45906.287581,45906.287581,45906.287581,52.696928,15.060680
2,2020,BGR,Bulgaria,901,"Coffee, whether or not roasted or decaffeinate...",3.998450e+07,1.313092e+08,Coffee,Coffee total,3.284003,6550696.0,10760.211975,10760.211975,10760.211975,20.045079,6.103855
3,2020,AUT,Austria,1801,"Cocoa beans; whole or broken, raw or roasted",2.658997e+07,7.387447e+07,Cocoa,Cocoa beans,2.778283,8916864.0,48716.409890,48716.409890,48716.409890,8.284803,2.981987
4,2020,BEL,Belgium,1801,"Cocoa beans; whole or broken, raw or roasted",8.124282e+07,2.325734e+08,Cocoa,Cocoa beans,2.862694,11538604.0,45906.287581,45906.287581,45906.287581,20.156110,7.040958


In [71]:
# Validación estadística de las métricas per cápita.
#
# Se revisan las principales métricas calculadas para detectar valores extremos,
# nulos o resultados poco coherentes antes de utilizarlas en el dashboard.

trade_enriched_complete[
    [
        "import_value_usd_per_capita",
        "net_weight_kg_per_capita",
        "gdp_per_capita"
    ]
].describe()

,import_value_usd_per_capita,net_weight_kg_per_capita,gdp_per_capita
count,405.000000,394.000000,405.000000
mean,35.395990,6.534233,40843.353398
std,35.599633,6.706734,26370.710242
min,0.000793,0.000000,10760.211975
25%,7.037544,1.703151,22710.263754
50%,29.587570,5.565258,31823.436393
75%,50.618366,9.970228,52834.291675
max,237.074098,45.436410,137781.681659


In [72]:
trade_enriched_complete.columns.tolist()


['year',
 'country_code',
 'country',
 'commodity_code',
 'commodity_description',
 'net_weight_kg',
 'import_value_usd',
 'sector',
 'product_group',
 'avg_price_usd_kg',
 'population',
 'gdp_per_capita_x',
 'gdp_per_capita_y',
 'gdp_per_capita',
 'import_value_usd_per_capita',
 'net_weight_kg_per_capita']

In [73]:
# Limpieza de columnas duplicadas tras los cruces.
#
# Durante el proceso de integración se generaron columnas duplicadas
# de PIB per cápita con sufijos _x e _y.
# Se conserva una única variable gdp_per_capita para evitar ambigüedades
# en el dataset final.

trade_enriched_complete = trade_enriched_complete.drop(
    columns=["gdp_per_capita_x", "gdp_per_capita_y"]
)

trade_enriched_complete.columns.tolist()

['year',
 'country_code',
 'country',
 'commodity_code',
 'commodity_description',
 'net_weight_kg',
 'import_value_usd',
 'sector',
 'product_group',
 'avg_price_usd_kg',
 'population',
 'gdp_per_capita',
 'import_value_usd_per_capita',
 'net_weight_kg_per_capita']

In [74]:
# Validación final de nulos en el dataset enriquecido.
#
# Esta revisión permite comprobar qué variables conservan valores nulos
# antes de exportar el dataset procesado.

trade_enriched_complete.isna().sum()

year                            0
country_code                    0
country                         0
commodity_code                  0
commodity_description           0
net_weight_kg                  11
import_value_usd                0
sector                          0
product_group                   0
avg_price_usd_kg               18
population                      0
gdp_per_capita                  0
import_value_usd_per_capita     0
net_weight_kg_per_capita       11
dtype: int64

In [76]:
# Exportación del dataset procesado.
#
# Se guarda el dataset final limpio en la carpeta data/processed.
# Este archivo será la base para el dashboard de Streamlit.

processed_path = Path("../data/processed/trade_enriched_complete_2020_2024.csv")

trade_enriched_complete.to_csv(
    processed_path,
    index=False
)

print("Archivo exportado en:", processed_path)

Archivo exportado en: ../data/processed/trade_enriched_complete_2020_2024.csv
